File: bihit-api/app/models/post.rb
class Post < ApplicationRecord
  belongs_to :user
  belongs_to :subcategory
  scope :subcategory_id, -> (subcategory_id) { where subcategory_id: subcategory_id }
  validates :title, presence: true, uniqueness: { scope: :subcategory_id, message: 'Already exists.' }
  validates :short, presence: true
  validates :long, presence: true
  has_many :comments, :dependent => :delete_all
  has_many :likes, :dependent => :delete_all
end

File: bihit-web/src/app/models/post.ts
import { Autor } from './autor';
export class Post {
    id: number;
    title: string;
    short: string;
    long: string;
    user_id: number;
    subcategory_id: number;
    createdAt: string;
    autor: Autor;
    likes: number;
    comments: number;

    constructor(id: number, title: string, short: string, long: string,
                createdAt: string, user_id: number, subcategory_id: number, autor: Autor, likes: number, comments: number)
    {
        this.id = id;
        this.title = title;
        this.createdAt = createdAt;
        this.short = short;
        this.long = long;
        this.user_id = user_id;
        this.subcategory_id = subcategory_id;
        this.autor = autor;
        this.likes = likes;
        this.comments = comments;
    }   
}